stan/math/prim/fun/squared_distance.hpp

```diff
@@ -12,13 +12,14 @@ namespace math {
 /**
  * Returns the squared distance.
  *
- * @param x1 First vector.
- * @param x2 Second vector.
- * @return Dot product of the vectors.
- * @throw std::domain_error If the vectors are not the same
- * size or if they are both not vector dimensioned.
+ * @param x1 First scalar.
+ * @param x2 Second scalar.
+ * @return Squared distance between scalars
+ * @throw std::domain_error Any scalar is not finite.
```

Comment:
Would you mind documenting T1 and T2 as well. 

[Optionally] Could also make the names something nicer like `Scalar1` and `Scalar2`

Reply:
done